tests/utils.py

```diff
@@ -7,16 +7,18 @@
 import subprocess
 import sys
 import venv
-from collections.abc import Mapping
+from collections.abc import Iterable, Mapping
 from functools import cache
 from pathlib import Path
-from typing import Iterable, NamedTuple
-from typing_extensions import Annotated
+from typing import TYPE_CHECKING, NamedTuple
 
 import pathspec  # type: ignore[import]
 import tomli
 from packaging.requirements import Requirement
 
+if TYPE_CHECKING:
```

Comment:
I feel like I prefer the original version here, where it wasn't guarded behind the `TYPE_CHECKING` flag. It's true that this avoids an import at runtime, but the performance benefit here is tiny, and I prefer the cleaner and simpler code :)

Having said that, I think `typing_extensions` is only available here because `mypy` depends on it, and we have `mypy` in our `requirements-tests.txt` file. We should probably add it to `requirements-tests.txt`, since implicit dependencies are generally a bad idea, and we also need the dependency for e.g. `stubtest_stdlib.py`. Want to make another PR adding it to `requirements-tests.txt`?

Reply:
Oh yeah this should've been reverted too when installing from `requirements-tests.txt`.

Sure I'll do a PR to add the dep explicitely.